infernyx/rules.py

```diff
@@ -231,6 +232,44 @@ def report_rule_stats(job):
         log.error("Error writing stats %s" % e)
 
 
+def report_suspicious_ips(it, params, job_id):
+    total_ips = 0
+    ips = []
+    for (_, ip), (total,) in it:
+        if total_ips >= 1000:
+            break
+        ips.append("%20s %d" % (ip, total))
+        total_ips += 1
+
+    if total_ips:
+        msg = '\n'.join(ips)
+        statsd.event("Suspicious IP report", msg)
```

Comment:
can you doublecheck the 8k limit for datadog?  that kind of sucks.

On Wed, May 13, 2015 at 5:17 PM, Nan Jiang <notifications@github.com> wrote:

> In infernyx/rules.py
> <https://github.com/tspurway/infernyx/pull/20#discussion_r30277231>:
>
> > @@ -231,6 +232,44 @@ def report_rule_stats(job):
> >          log.error("Error writing stats %s" % e)
> >
> >
> > +def report_suspicious_ips(it, params, job_id):
> > +    total_ips = 0
> > +    ips = []
> > +    for (_, ip), (total,) in it:
> > +        if total_ips >= 1000:
> > +            break
> > +        ips.append("%20s %d" % (ip, total))
> > +        total_ips += 1
> > +
> > +    if total_ips:
> > +        msg = '\n'.join(ips)
> > +        statsd.event("Suspicious IP report", msg)
>
> The statsd.event() will raise an exception if the length of msg is
> greater than 8k. We'd better drop an exception handler here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tspurway/infernyx/pull/20/files#r30277231>.
>